src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java

```diff
@@ -210,12 +213,12 @@ public Config validate(Map<String, String> connectorConfigs) {
     }
 
     // If using snowflake_jwt and authentication, and private key or private key passphrase is
-    // provided through file, skip validation
+    // provided through a config provider, skip validation
     if (connectorConfigs
             .getOrDefault(Utils.SF_AUTHENTICATOR, Utils.SNOWFLAKE_JWT)
             .equals(Utils.SNOWFLAKE_JWT)
-        && (connectorConfigs.getOrDefault(Utils.SF_PRIVATE_KEY, "").contains("${file:")
-            || connectorConfigs.getOrDefault(Utils.PRIVATE_KEY_PASSPHRASE, "").contains("${file:")))
+        && (configProviderPrefix.matcher(connectorConfigs.getOrDefault(Utils.SF_PRIVATE_KEY, "")).find()
```

Comment:
I would rewrite this if statement to be a little bit more readable. E.g:
```
if (usingJWT(connectorConfigs) && keyMatchesPattern(connectorConfigs)) {
```

Another benefit of doing so will be moving the pattern to the method that performs actual matching.